app/storage/metadata_parser.py

```diff
@@ -1,103 +1,108 @@
+    This custom field deserialises UUIDs to strings.
+    """
 
-def optional_string_parser(optional_string_value):
-    if not isinstance(optional_string_value, str):
-        raise TypeError('Claim not a valid/empty string')
-    return optional_string_value
+    def _deserialize(self, *args, **kwargs):  # pylint: disable=arguments-differ
+        return str(super()._deserialize(*args, **kwargs))
 
 
-def clean_leading_trailing_spaces(metadata):
-    for key, value in metadata.items():
-        if isinstance(value, str):
-            metadata[key] = value.strip()
+class DateString(fields.DateTime):
+    """ Currently, runner cannot handle Date objects in metadata
+    Since all metadata is serialised and deserialised to JSON.
+    This custom field deserialises Dates to strings.
+    """
 
-    return metadata
+    def _deserialize(self, *args, **kwargs):  # pylint: disable=arguments-differ
+        return super()._deserialize(*args, **kwargs).strftime('%Y-%m-%d')
 
 
 VALIDATORS = {
-    'date': iso_8601_date_parser,
-    'uuid': uuid_4_parser,
-    'boolean': boolean_parser,
-    'string': string_parser,
-    'optional_string': optional_string_parser,
+    'date': functools.partial(DateString, format='%Y-%m-%d', required=True),
+    'uuid': functools.partial(UUIDString, required=True),
+    'boolean': functools.partial(fields.Boolean, required=True),
+    'string': functools.partial(fields.String, required=True),
+    'url': functools.partial(fields.Url, required=True),
 }
 
-MANDATORY_METADATA = [
-    {'name': 'eq_id', 'validator': 'string'},
-    {'name': 'form_type', 'validator': 'string'},
-    {'name': 'ru_ref', 'validator': 'string'},
-    {'name': 'collection_exercise_sid', 'validator': 'string'},
-    {'name': 'tx_id', 'validator': 'uuid'},
-    {'name': 'case_id', 'validator': 'uuid'},
-    {'name': 'response_id', 'validator': 'string'},
-]
 
+class StripWhitespaceMixin:
+    @pre_load
+    def strip_whitespace(self, items):  # pylint: disable=no-self-use
+        for key, value in items.items():
+            if isinstance(value, str):
+                items[key] = value.strip()
+        return items
 
-def parse_runner_claims(claims):
-    cleaned_claims = clean_leading_trailing_spaces(claims.copy())
-    validate_metadata(cleaned_claims, MANDATORY_METADATA)
 
-    return cleaned_claims
+class RunnerMetadataSchema(Schema, StripWhitespaceMixin):
+    """ Metadata which is required for the operation of runner itself
+    """
 
+    jti = VALIDATORS['uuid']()
+    eq_id = VALIDATORS['string'](validate=validate.Length(min=1))
+    form_type = VALIDATORS['string'](validate=validate.Length(min=1))
+    ru_ref = VALIDATORS['string'](validate=validate.Length(min=1))
+    collection_exercise_sid = VALIDATORS['string'](validate=validate.Length(min=1))
+    tx_id = VALIDATORS['uuid']()
+    case_id = VALIDATORS['uuid']()
+    response_id = VALIDATORS['string']()
+
+    # Account Service URL should be mandatory
```

Comment:
Shouldn't?

Reply:
No, it should be required, but I had to make it optional.

Reply:
Apparently I hadn't thought about this much. It should have been mandatory and could have been mandatory. 

It's now mandatory.